### PR TITLE
Added Oj gem to use as JSON serializer/deserializer.

### PR DESCRIPTION
**Pull request by mark-jordanovic-lewis**
Features:
Improve JSON parsing in dvelp_api_response

**Change**

Change Owners: Mark Jordanovic-Lewis
Reason: Faster parsing -> Shorter response time -> More people served -> Customers happier

**Description**

Netflix fast_jsonapi touts itself as a rapid JSON loading and generating library, it is desired to approach the speeds offered by this. Reasons for minimising development lift are described in ticket. In short, the approach of dvelp_api_response is similar with less validation logic this is due to dvelp_api_response being simpler rather than less safe. Choice of JSON (de)serialiser implied a quick win in the stated desire of the ticket.

ticket: https://trello.com/c/Mfi5Doit

Validation

- specs pass and additional specs added to document changes in JSON parse error messages.
- **todo**: Local testing via postman of airline-api-manager and airline-api with dvelp_api_response.
- **todo**: Each app that leverages dvelp_api_response updated and deployed to staging and manual tested for Twilio and Twilio Flex functionality compatibility. Gemfiles pointed at github branch version of the gem.
- Observe metrics around response times, hopefully these should drop. 

**Pre-deploy Plan**
- Add dependencies in airline-api and airline-manager-api for the new gem version via github branch.
- Deploy apps to staging and see if they are stable.
- Add dependencies in all other apps which use dvelp_api_response gem and leave to see if they are stable - this will require the co-operation of devs working on related projects and what they have currently deployed.
- If all remains kosher then the Go Live Plan can be moved on to.

**Go Live Plan**
- Build and deploy dvelp_api_reponse v 0.7.2 to gemfury
- Update dependecies in all dependent apps, merge to master and deploy one by one

**Rollback plan**
- dvelp_api_responce v 0.7.1 still available on Gemfury. Rollback app Gemfiles to that and redeploy.

**Risks**
- After testing on staging to ensure response coherence and subsequent fixes there should be no issue.

**Monitoring**
_Active_
Assigned: MarkJL
- Manual observation of Sentry for an hour
- Monitor metrics around response times

_Passive_
- Watch for sentry errors in slack
- Look at metrics around response times - these _should not_ increase.

**Changes**

- Added specs to cover unfriendly error messages
- Used Oj.load and Oj.dump in place of JSON.parse and x.to_json
- Used new FactoryBot value assign syntax
- Bumped bundler version to 1.17.3
- Removed sprocket and view loading from combustion gem

### CODE REVIEW

File: lib/dvelp_api_response/api_request_helper.rb [hunk 1]
```diff
@@ -63,5 +63,13 @@ def fill_timestamp(timestamp)
     def fill_accept(media_type)
       request.set_header 'HTTP_ACCEPT', Mime[media_type].to_s
     end
+
+    def jsonify_payload(hash)
+      Oj.dump(hash, mode: :rails, time_format: :ruby)
```
Comment (mark-jordanovic-lewis) [inline lib/dvelp_api_response/api_request_helper.rb]: This uses `Time/Date/DateTime`'s `.to_s` to generate the time string. Not sure that's what we want but the time format can be set to most formats.

File: config/initializers/oj.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+require 'oj'
+
+Oj::Rails.set_encoder()
```
Comment (tom-mullen) [inline config/initializers/oj.rb]: Mark, what's the thought process behind using OJ over the fast_json_api gem? Or will we use both?

Comment (mark-jordanovic-lewis) [reply config/initializers/oj.rb]: I've recorded my thoughts in the ticket: https://trello.com/c/Mfi5Doit and in short in the description.

tl;dr: fast_jsonapi gem is _not_ a json serialisation gem itself. It leverages serialisation gems and provides a nice interface to AR models. The approach used in dvelp_api_response is consistent but simpler than that in fast_jsonapi.

Comment (mark-jordanovic-lewis) [reply config/initializers/oj.rb]: Result of this conversation has been recorded on the ticket and spawned two new tickets. 
- N+1 queries to be removed: https://trello.com/c/NOPgdV6I
- Caching to be added: https://trello.com/c/McOPhN3A
